File: Ruby/Euler001.rb
#!/usr/bin/env ruby

def main
  sum = (3..999).step(3).sum { |x| x }

  sum += (5..999).step(5).filter { |x| x % 3 != 0 }.sum

  puts sum
end

if __FILE__ == $0
  main
end

File: Python/Euler001.py
#!/usr/bin/env python3
from __future__ import print_function
import sys
if sys.version.startswith('2'):
    range = xrange


def main():
    s = sum(range(3, 1000, 3)) + sum(i for i in range(5, 1000, 5) if i % 3)
    print(s)


if __name__ == '__main__':
    main()
